src/callbacks_stage/subcell_bounds_check.jl

```diff
@@ -0,0 +1,124 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+"""
+    BoundsCheckCallback(; output_directory="out", save_errors=false, interval=1)
+
+Bounds checking routine for [`SubcellLimiterIDP`](@ref). Applied as a stage callback for SSPRK
+methods. If `save_errors` is `true`, the resulting deviations are saved in
+`output_directory/deviations.txt` for every `interval` time steps.
+
+!!! note
+    For `SubcellLimiterIDP`, the solution is corrected in the a posteriori correction stage
+    [`SubcellLimiterIDPCorrection`](@ref). So, to check the final solution, this bounds check
+    callback must be called after the correction stage.
+"""
+struct BoundsCheckCallback
+    output_directory::String
+    save_errors::Bool
+    interval::Int
+end
+
+function BoundsCheckCallback(; output_directory = "out", save_errors = false,
+                             interval = 1)
+    BoundsCheckCallback(output_directory, save_errors, interval)
+end
+
+function (callback::BoundsCheckCallback)(u_ode, integrator, stage)
+    mesh, equations, solver, cache = mesh_equations_solver_cache(integrator.p)
+    (; t, iter, alg) = integrator
+    u = wrap_array(u_ode, mesh, equations, solver, cache)
+
+    save_errors_ = callback.save_errors && (callback.interval > 0) &&
+                   (stage == length(alg.c)) &&
+                   (iter % callback.interval == 0 || integrator.finalstep)
+    @trixi_timeit timer() "check_bounds" check_bounds(u, mesh, equations, solver, cache,
+                                                      solver.volume_integral, t,
+                                                      iter + 1,
+                                                      callback.output_directory,
+                                                      save_errors_)
```

Comment:
```suggestion
    save_errors = callback.save_errors && (callback.interval > 0) &&
                  (stage == length(alg.c)) &&
                  (iter % callback.interval == 0 || integrator.finalstep)
    @trixi_timeit timer() "check_bounds" check_bounds(u, mesh, equations, solver, cache,
                                                      solver.volume_integral, t,
                                                      iter + 1,
                                                      callback.output_directory,
                                                      save_errors)
```
Why the trailing `_`?

Reply:
Good point. Don't need it. Thanks.